plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -89,7 +89,42 @@ func LeastRequestedPriority(pod *api.Pod, podLister algorithm.PodLister, minionL
 
 	list := algorithm.HostPriorityList{}
 	for _, node := range nodes.Items {
-		list = append(list, calculateOccupancy(pod, node, podsToMachines[node.Name]))
+		list = append(list, calculateResourceOccupancy(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func min(l, r int64) (m int64) {
+	m = r
+	if l < r {
+		m = l
+	}
+	return m
+}
+
+// See comment for DumbSpreadingPriority()
+const dumbSpreadingDenominator int64 = 10
+
+// DumbSpreadingPriority is a priority function that favors nodes with fewer pods.
+// It works like LeastRequestedPeriority but instead of using 10 * percentage of machine free by resource,
+// it uses 10 * percentage of machine free by pod, with "percentage of machine free by pod" claculated as
+// (dumbSpreadingDenominator - number of pods already on the node + 1) / dumbSpreadingDenominator.
+// dumbSpreadingDenominator serves like the machine capacity in LeasRequestedPriority but is chosen
+// so that we equate one pod with a reasonable amount of resources when we combine all the scores together.
```

Comment:
I agree with your observation that in the current version, once you reach 10 (or more) pods, DumbSpreadPriority == 0 and we schedule only based on resources. We could make the denominator larger, which would cause the DumbSpreadPriority to decline more slowly as # pods increases, but  that can also lead to giving too much weight to the resources (limit).

For example, today a machine that's 80% full with 1 pod has the same score (call it 2 + 8 = 10) as a machine that's 20% full with 7 pods (8 + 2). This feels vaguely in the right ballpark, I guess. But say we make the denominator 30. Now the first machine has a score of 2 + 9.33 = 11.33 and the second machine has a score of 8 + 7.33 = 15.33 so we're strongly favoring the second machine--we're giving too much weight to limit, and not enough to number of pods. In fact they won't have an equal score until there are 20 pods on the second machine. (8 + 3.33). That seems bad.

I'm open to suggestions on how to address the problem you brought up. Is there a different calculateScore() function we could use? It's too late at night for me to think of one right now.